compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -393,10 +393,12 @@ public void visit(BLangXMLNSStatement xmlnsStmtNode, AnalyzerData data) {
     @Override
     public void visit(BLangResourceFunction funcNode, AnalyzerData data) {
         visit((BLangFunction) funcNode, data);
+        BType returnType = funcNode.returnTypeNode.getBType();
+        checkForClientObjectTypeOrFunctionType(returnType, funcNode.returnTypeNode.getPosition());
```

Comment:
:pick: You can now provide the error outside like your first implementation and you don't have to pass the position in that way.

Reply:
Updated ed32dfec08200b69841e49a801715f0360027da2